src/main/java/teammates/storage/api/NotificationsDb.java

```diff
@@ -37,6 +42,61 @@ public NotificationAttributes getNotification(String notificationId) {
         return notificationId.isEmpty() ? null : makeAttributesOrNull(getNotificationEntity(notificationId));
     }
 
+    /**
+     * Gets all notifications.
+     */
+    public List<NotificationAttributes> getAllNotifications() {
+        List<Notification> notifications = load().list();
+        List<NotificationAttributes> notificationAttributes = makeAttributes(notifications);
+        NotificationAttributes.sortByStartTime(notificationAttributes);
+        return notificationAttributes;
+    }
+
+    /**
+     * Gets notifications by {@code targetUser}.
+     *
+     * @return a list of notifications for the specified targetUser.
+     */
+    public List<NotificationAttributes> getNotificationsByTargetUser(NotificationTargetUser targetUser) {
+        assert targetUser != null;
+
+        List<Notification> endEntities = load()
+                .filter("targetUser", targetUser)
+                .filter("endTime >", Instant.now())
+                .list();
+
+        List<Notification> startEntities = load()
+                .filter("targetUser", targetUser)
+                .filter("startTime <", Instant.now())
+                .list();
+
+        if (targetUser != NotificationTargetUser.GENERAL) {
+            endEntities.addAll(load()
+                    .filter("targetUser", NotificationTargetUser.GENERAL)
+                    .filter("endTime >", Instant.now())
+                    .list());
+            startEntities.addAll(load()
+                    .filter("targetUser", NotificationTargetUser.GENERAL)
+                    .filter("startTime <", Instant.now())
+                    .list());
+        }
+
+        List<String> startEntitiesIds = startEntities.stream()
+                .map(Notification::getNotificationId)
+                .collect(Collectors.toList());
+
+        List<Notification> ongoingNotifications = endEntities.stream()
+                .filter(notif -> {
+                    String id = notif.getNotificationId();
+                    return startEntitiesIds.contains(id);
+                })
+                .collect(Collectors.toList());
```

Comment:
Same comment as above. Would it be better to use two filter methods so we don't have to do these additional processing?